Felix search function

FelixHultman opened the pull request:
Added a searchfunction in the navbar. 
when a movie is picked in the searchbar user will re redirected to a movie page based on movie-id. 
i had to change the model.ts a bit for it to work, take a look so i did not break anything. 
everything seems to work fine when i merged the main into my branch but take a look if there is something i missed.

JrSchoolAccount commented:
Looks really good, great work!

It seems to work backwards.
when i go to type in what i want to search it shows all movies and then removes them to reflect my search :)

Other than that it works great.

Maybe the function can be implemented with a onChange function.
``` tsx
  const [search, setSearch] = React.useState('');

<TextField
    id='search'
    name='search'
    value={search}
    onChange={(e) => setSearch(e.target.value)}
/>
```

NelsonTheBrave commented:
We didn't discuss exactly how the search function would work, I believe this is just one way to do it that works in this context. I don't think you've broken anything in models.ts :)
The only comment I have is to change the language to Swedish in the placeholder text of the search window, otherwise great!

FelixHultman commented:
@JrSchoolAccount @NelsonTheBrave 
thanks for input. i've changed the function to Jesper suggestion, the way it worked is fine for when we only have this amount of movies but with a larger collection it would be kinda bad i guess. Placeholder text is also changed!

NelsonTheBrave commented on app/ui/navbar.tsx:
```diff
@@ -13,20 +13,51 @@ import Avatar from '@mui/material/Avatar';
+          const movieData = data.movies.map((movie: any) => ({
+            title: movie.title,
+            id: movie._id,
+            link: `/movies/${movie._id}`,
```
Here is one place where it should be filmer instead of movies

FelixHultman replied:
that's actually dead code, so thanks for telling me!

JrSchoolAccount reviewed (commented):
Looks really good!
I like the new changes to the code :)

I dont really like the useEffect tbh.

Maybe you can have the searchbar function with a onChange instead of useEffect.
Heres an example:

```ts
const handleSearchInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    const filtered = movies.filter((movie) =>
      movie.title.toLowerCase().includes(value.toLowerCase())
    );
    setFilteredMovies(filtered);
  };

  const handleSearchSelect = (value: Movie | string | null) => {
    if (value && typeof value !== 'string') {
      return `/filmer/${value.id}`;
    }
    return '';
  };

<Stack spacing={5} sx={{ width: 300, marginRight: 5 }}>
            <Autocomplete
              key="movieSearchAutocomplete"
              freeSolo
              id="movieSearch"
              disableClearable
              options={filteredMovies}
              getOptionLabel={(option) => {
                if (typeof option === 'string') {
                  return option;
                }
                return option.title;
              }}
              onChange={(event, value) => {
                const movieUrl = handleSearchSelect(value);
                if (movieUrl) {
                  window.location.href = movieUrl;
                }
              }}
              renderInput={(params) => (
                <TextField
                  {...params}
                  label="Sök film.."
                  InputProps={{
                    ...params.InputProps,
                    type: 'search',
                    onChange: handleSearchInputChange,
                  }}
                />
              )}
            />
          </Stack>
```

NelsonTheBrave reviewed (approved):
From what I can see you have implemented all the suggested changes and it works well and as intended! Good job!
